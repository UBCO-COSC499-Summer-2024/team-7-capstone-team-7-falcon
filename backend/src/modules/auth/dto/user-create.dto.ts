import {
  IsEmail,
  IsInt,
  IsNotEmpty,
  IsOptional,
  IsString,
  IsStrongPassword,
  Length,
  Min,
} from 'class-validator';
import { ERROR_MESSAGES } from '../../../common';
import { MatchPasswords } from '../../../decorators/match-passwords.decorator';

export class UserCreateDto {
  @IsString({ message: ERROR_MESSAGES.userEditDto.firstNameString })
  @IsNotEmpty({ message: ERROR_MESSAGES.userEditDto.firstNameRequired })
  @Length(2, 15, { message: ERROR_MESSAGES.userEditDto.firstNameLength })
  readonly first_name!: string;

  @IsOptional()
  readonly last_name?: string;

  @IsEmail()
  @IsNotEmpty({ message: ERROR_MESSAGES.common.emailRequired })
  readonly email!: string;

  @IsString({ message: ERROR_MESSAGES.common.passwordString })
  @IsNotEmpty({ message: ERROR_MESSAGES.common.passwordRequired })
  @IsStrongPassword(
    {
      minLength: 8,
      minLowercase: 1,
      minUppercase: 1,
      minNumbers: 1,
      minSymbols: 1,
    },
    { message: ERROR_MESSAGES.common.passwordStrong },
  )
  readonly password!: string;

  @IsString({ message: ERROR_MESSAGES.common.confirmPasswordString })
  @IsNotEmpty({ message: ERROR_MESSAGES.common.confirmPasswordRequired })
  @MatchPasswords('password', {
    message: ERROR_MESSAGES.common.passwordMustMatch,
  })
  readonly confirm_password!: string;

  @IsInt({ message: ERROR_MESSAGES.userEditDto.studentNumberMustBeNumber })
  @Min(1, { message: ERROR_MESSAGES.userEditDto.studentNumberMustBeGreater })
  @IsOptional()
  readonly student_id?: number;

  @IsInt({ message: ERROR_MESSAGES.userEditDto.employeeNumberMustBeNumber })
  @Min(1, { message: ERROR_MESSAGES.userEditDto.employeeNumberMustBeGreater })
  @IsOptional()
  readonly employee_id?: number;
}
