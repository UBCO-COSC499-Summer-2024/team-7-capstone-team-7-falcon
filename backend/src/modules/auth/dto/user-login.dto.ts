import { IsEmail, IsNotEmpty, IsString } from 'class-validator';
import { ERROR_MESSAGES } from '../../../common';

export class UserLoginDto {
  @IsEmail()
  @IsNotEmpty({ message: ERROR_MESSAGES.common.emailRequired })
  email!: string;

  @IsString({ message: ERROR_MESSAGES.common.passwordString })
  @IsNotEmpty({ message: ERROR_MESSAGES.common.passwordRequired })
  password!: string;
}
