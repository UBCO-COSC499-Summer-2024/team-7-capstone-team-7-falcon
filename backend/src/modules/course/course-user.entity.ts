import { CourseRoleEnum } from '../../enums/user.enum';
import {
  Entity,
  PrimaryGeneratedColumn,
  Column,
  BaseEntity,
  ManyToOne,
  JoinColumn,
} from 'typeorm';
import { UserModel } from '../user/entities/user.entity';
import { CourseModel } from './course.entity';

@Entity('course_user_model')
export class CourseUser extends BaseEntity {
  @PrimaryGeneratedColumn()
  id: number;

  @Column({ default: CourseRoleEnum.STUDENT })
  course_role: CourseRoleEnum;

  @ManyToOne(() => UserModel, (user) => user.courses)
  @JoinColumn({ name: 'user_id' })
  user: UserModel;

  @ManyToOne(() => CourseModel, (course) => course.users)
  @JoinColumn({ name: 'course_id' })
  course: CourseModel;
}
