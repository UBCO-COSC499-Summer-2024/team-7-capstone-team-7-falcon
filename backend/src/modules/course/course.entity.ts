import {
  Entity,
  PrimaryGeneratedColumn,
  Column,
  BaseEntity,
  OneToMany,
} from 'typeorm';
import { CourseUser } from './course-user.entity';
import { ExamModel } from '../exams/entities/exam.entity';

@Entity('course_model')
export class CourseModel extends BaseEntity {
  @PrimaryGeneratedColumn()
  id: number;

  @Column()
  course_code: string;

  @Column()
  course_name: string;

  @Column({ type: 'bigint' })
  created_at: number;

  @Column({ type: 'bigint' })
  updated_at: number;

  // add term id later

  @Column({ default: false })
  is_archived: boolean;

  @Column()
  invite_code: string;

  @Column()
  section_name: string;

  @OneToMany(() => CourseUser, (user) => user.course)
  users: CourseUser[];

  @OneToMany(() => ExamModel, (exam) => exam.course)
  exams: ExamModel[];
}
