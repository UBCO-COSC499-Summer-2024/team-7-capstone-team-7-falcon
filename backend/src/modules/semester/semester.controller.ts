import {
  Body,
  Controller,
  Get,
  HttpStatus,
  Param,
  ParseIntPipe,
  Patch,
  Post,
  Res,
  UseGuards,
  ValidationPipe,
} from '@nestjs/common';
import { AuthGuard } from '../../guards/auth.guard';
import { SystemRoleGuard } from '../../guards/system-role.guard';
import { SemesterDto } from './dto/semester.dto';
import { Roles } from '../../decorators/roles.decorator';
import { UserRoleEnum } from '../../enums/user.enum';
import { SemesterService } from './semester.service';
import { Response } from 'express';
import {
  SemesterCreationException,
  SemesterNotFoundException,
} from '../../common/errors';
import { ERROR_MESSAGES } from '../../common';

@Controller('semester')
export class SemesterController {
  /**
   * Constructor of SemesterController
   * @param semesterService {SemesterService} - The semester service
   */
  constructor(private readonly semesterService: SemesterService) {}

  /**
   * Get all semesters
   * @param res {Response} - The response object
   * @returns {Promise<Response>} - The response object with semesters or error message
   */
  @UseGuards(AuthGuard)
  @Get('/limited/all')
  async getAllLimited(@Res() res: Response): Promise<Response> {
    const semesters = await this.semesterService.getAllSemestersLimited();

    if (!semesters) {
      return res.status(HttpStatus.NOT_FOUND).send({
        message: ERROR_MESSAGES.semesterController.semestersNotFound,
      });
    }

    return res.status(HttpStatus.OK).send(semesters);
  }

  /**
   * Get all semesters
   * @param res {Response} - The response object
   * @returns {Promise<Response>} - The response object with semesters or error message
   */
  @UseGuards(AuthGuard, SystemRoleGuard)
  @Roles(UserRoleEnum.ADMIN)
  @Get('/all')
  async getAll(@Res() res: Response): Promise<Response> {
    try {
      const semesters = await this.semesterService.getAllSemesters();

      if (semesters.length === 0) {
        return res.status(HttpStatus.NO_CONTENT).send();
      }

      return res.status(HttpStatus.OK).send(semesters);
    } catch (e) {
      return res.status(HttpStatus.INTERNAL_SERVER_ERROR).send({
        message: e.message,
      });
    }
  }

  /**
   * Create a new semester
   * @param res {Response} - The response object
   * @param body {SemesterCreateDto} - The semester details
   * @returns {Promise<Response>} - The response object
   */
  @UseGuards(AuthGuard, SystemRoleGuard)
  @Roles(UserRoleEnum.ADMIN)
  @Post('/create')
  async post(
    @Res() res: Response,
    @Body(new ValidationPipe()) body: SemesterDto,
  ): Promise<Response> {
    try {
      await this.semesterService.createSemester(body);
      return res.status(HttpStatus.CREATED).send({
        message: 'ok',
      });
    } catch (e) {
      if (e instanceof SemesterCreationException) {
        return res.status(HttpStatus.BAD_REQUEST).send({
          message: e.message,
        });
      } else {
        return res.status(HttpStatus.INTERNAL_SERVER_ERROR).send({
          message: e.message,
        });
      }
    }
  }

  /**
   * Update semester details by id
   * @param res {Response} - The response object
   * @param sid {number} - The semester id {sid
   * @param body {SemesterDto} - The semester details {body
   * @returns {Promise<Response>} - The response object
   */
  @UseGuards(AuthGuard, SystemRoleGuard)
  @Roles(UserRoleEnum.ADMIN)
  @Patch('/:sid')
  async update(
    @Res() res: Response,
    @Param('sid', ParseIntPipe) sid: number,
    @Body(new ValidationPipe()) body: SemesterDto,
  ): Promise<Response> {
    try {
      await this.semesterService.updateSemester(sid, body);
      return res.status(HttpStatus.NO_CONTENT).send();
    } catch (e) {
      if (e instanceof SemesterNotFoundException) {
        return res.status(HttpStatus.NOT_FOUND).send({
          message: e.message,
        });
      } else if (e instanceof SemesterCreationException) {
        return res.status(HttpStatus.BAD_REQUEST).send({
          message: e.message,
        });
      } else {
        return res.status(HttpStatus.INTERNAL_SERVER_ERROR).send({
          message: e.message,
        });
      }
    }
  }

  /**
   * Get semester by id
   * @param res {Response} - The response object {res
   * @param sid {number} - The semester id
   * @returns {Promise<Response>} - The response object
   */
  @UseGuards(AuthGuard, SystemRoleGuard)
  @Roles(UserRoleEnum.ADMIN)
  @Get('/:sid')
  async getSemesterById(
    @Res() res: Response,
    @Param('sid', ParseIntPipe) sid: number,
  ): Promise<Response> {
    try {
      const semester = await this.semesterService.getSemesterById(sid);
      return res.status(HttpStatus.OK).send(semester);
    } catch (e) {
      if (e instanceof SemesterNotFoundException) {
        return res.status(HttpStatus.NOT_FOUND).send({
          message: e.message,
        });
      } else {
        return res.status(HttpStatus.INTERNAL_SERVER_ERROR).send({
          message: e.message,
        });
      }
    }
  }
}
