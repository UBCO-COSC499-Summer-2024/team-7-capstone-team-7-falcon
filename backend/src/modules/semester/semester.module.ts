import { Module } from '@nestjs/common';
import { SemesterController } from './semester.controller';
import { SemesterService } from './semester.service';
import { UserService } from '../user/user.service';
import { TokenService } from '../token/token.service';

@Module({
  controllers: [SemesterController],
  providers: [SemesterService, UserService, TokenService],
  exports: [SemesterService],
})
export class SemesterModule {}
