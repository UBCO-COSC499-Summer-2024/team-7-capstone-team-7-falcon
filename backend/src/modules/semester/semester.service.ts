import { Injectable } from '@nestjs/common';
import { SemesterDto } from './dto/semester.dto';
import {
  SemesterCreationException,
  SemesterNotFoundException,
} from '../../common/errors';
import { ERROR_MESSAGES } from '../../common';
import { SemesterModel } from './entities/semester.entity';
import { MoreThan } from 'typeorm';
import { CourseModel } from '../course/entities/course.entity';

@Injectable()
export class SemesterService {
  private readonly TWO_DAYS = 2 * 24 * 60 * 60 * 1000;
  private readonly THREE_MONTHS = 3 * 30 * 24 * 60 * 60 * 1000;

  /**
   * Get all semesters
   * @returns {Promise<SemesterModel[]>} - The promise object
   */
  public async getAllSemestersLimited(): Promise<SemesterModel[]> {
    const dateThreeMonthsAgo: number =
      parseInt(new Date().getTime().toString()) - this.THREE_MONTHS;

    const result = await SemesterModel.find({
      where: { starts_at: MoreThan(dateThreeMonthsAgo) },
      select: ['id', 'name'],
    });

    if (!result || result.length === 0) {
      return null;
    }

    return result;
  }

  /**
   * Get all semesters and their course count
   * @returns {Promise<SemesterModel[]>} - The promise object
   */
  public async getAllSemesters(): Promise<SemesterModel[]> {
    const semesters = await SemesterModel.createQueryBuilder('semester')
      .loadRelationCountAndMap('semester.course_count', 'semester.courses')
      .getMany();

    return semesters;
  }

  /**
   * Create a new semester
   * @param semesterDetails {SemesterDto} - The semester details
   * @returns {Promise<void>} - The promise object
   */
  public async createSemester(semesterDetails: SemesterDto): Promise<void> {
    if (semesterDetails.starts_at >= semesterDetails.ends_at) {
      throw new SemesterCreationException(
        ERROR_MESSAGES.semesterController.semesterStartDateMustBeBeforeEndDate,
      );
    }

    const currentDate: number = parseInt(new Date().getTime().toString());

    if (semesterDetails.starts_at <= currentDate - this.TWO_DAYS) {
      throw new SemesterCreationException(
        ERROR_MESSAGES.semesterController.semesterStartDateMustBeTwoDaysAhead,
      );
    }

    await SemesterModel.create({
      ...semesterDetails,
      created_at: currentDate,
      updated_at: currentDate,
    }).save();

    return;
  }

  /**
   * Update a semester
   * @param sid {number} - The semester id
   * @param semesterDetails {SemesterDto} - The semester details
   * @returns {Promise<void>} - The promise object
   */
  public async updateSemester(
    sid: number,
    semesterDetails: SemesterDto,
  ): Promise<void> {
    const semester = await SemesterModel.findOne({ where: { id: sid } });

    if (!semester) {
      throw new SemesterNotFoundException();
    }

    if (semesterDetails.starts_at >= semesterDetails.ends_at) {
      throw new SemesterCreationException(
        ERROR_MESSAGES.semesterController.semesterStartDateMustBeBeforeEndDate,
      );
    }

    await SemesterModel.update(sid, {
      ...semesterDetails,
      updated_at: parseInt(new Date().getTime().toString()),
    });

    return;
  }

  /**
   * Get a semester by id
   * @param sid {number} - The semester id
   * @returns {Promise<SemesterModel>} - The promise object
   */
  public async getSemesterById(sid: number): Promise<SemesterModel> {
    const semester = await SemesterModel.findOne({ where: { id: sid } });

    if (!semester) {
      throw new SemesterNotFoundException();
    }

    return semester;
  }

  /**
   * Delete a semester
   * @param sid {number} - The semester id
   * @returns {Promise<void>} - The promise object
   */
  public async deleteSemester(sid: number): Promise<void> {
    const semester = await this.getSemesterById(sid);

    await CourseModel.update({ semester: semester }, { semester: null });

    await semester.remove();
  }
}
