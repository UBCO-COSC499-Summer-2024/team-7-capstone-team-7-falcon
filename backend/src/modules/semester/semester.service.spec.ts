import { Test, TestingModule } from '@nestjs/testing';
import { SemesterService } from './semester.service';
import {
  TestConfigModule,
  TestTypeOrmModule,
} from '../../../test/utils/testUtils';
import { SemesterDto } from './dto/semester.dto';
import { SemesterModel } from './entities/semester.entity';
import { CourseModel } from '../course/entities/course.entity';

describe('SemesterService', () => {
  let semesterService: SemesterService;
  let moduleRef: TestingModule;

  beforeEach(async () => {
    moduleRef = await Test.createTestingModule({
      providers: [SemesterService],
      imports: [TestTypeOrmModule, TestConfigModule],
    }).compile();

    semesterService = moduleRef.get<SemesterService>(SemesterService);
  });

  afterEach(async () => {
    await moduleRef.close();
  });

  describe('getAllSemestersLimited', () => {
    it('should return all semesters', async () => {
      const currentDate: number = parseInt(new Date().getTime().toString());

      await SemesterModel.create({
        name: 'Test Semester',
        created_at: currentDate,
        updated_at: currentDate,
        starts_at: currentDate,
        ends_at: currentDate,
      }).save();

      const semesters = await semesterService.getAllSemestersLimited();
      expect(semesters).toBeDefined();
    });

    it('should return semesters that are within the last three months', async () => {
      const currentDate: number = parseInt(new Date().getTime().toString());
      const threeMonthsAgoDate: number =
        currentDate - 1000 * 60 * 60 * 24 * 30 * 3;

      await SemesterModel.create({
        name: 'Test Semester',
        created_at: currentDate,
        updated_at: currentDate,
        starts_at: currentDate,
        ends_at: currentDate,
      }).save();

      await SemesterModel.create({
        name: 'Test Semester 2',
        created_at: threeMonthsAgoDate,
        updated_at: threeMonthsAgoDate,
        starts_at: threeMonthsAgoDate,
        ends_at: threeMonthsAgoDate,
      }).save();

      const semesters = await semesterService.getAllSemestersLimited();

      expect(semesters).toHaveLength(1);
    });

    it('should return null if no semesters are found', async () => {
      const semesters = await semesterService.getAllSemestersLimited();

      expect(semesters).toBeNull();
    });
  });

  describe('createSemester', () => {
    it('should throw an error if the start date is after the end date', async () => {
      const semesterDetails: SemesterDto = {
        name: 'Test Semester',
        starts_at: parseInt(new Date('2021-01-01').getTime().toString()),
        ends_at: parseInt(new Date('2020-01-01').getTime().toString()),
      };

      await expect(
        semesterService.createSemester(semesterDetails),
      ).rejects.toThrow('Start date must be before end date');
    });

    it('should throw an error if the start date is behind two days', async () => {
      const semesterDetails: SemesterDto = {
        name: 'Test Semester',
        starts_at: parseInt(new Date('2021-01-01').getTime().toString()),
        ends_at: parseInt(new Date('2021-01-10').getTime().toString()),
      };

      await expect(
        semesterService.createSemester(semesterDetails),
      ).rejects.toThrow('Start date must be at least two days ahead');
    });

    it('should create a semester', async () => {
      const currentDate: number = parseInt(new Date().getTime().toString());
      const sevenDaysAheadDate: number = currentDate + 1000 * 60 * 60 * 24 * 7;

      const semesterDetails: SemesterDto = {
        name: 'Test Semester',
        starts_at: currentDate,
        ends_at: sevenDaysAheadDate,
      };

      await semesterService.createSemester(semesterDetails);

      const semester = await SemesterModel.findOne({
        where: { name: semesterDetails.name },
      });

      expect(semester).toBeDefined();

      expect(semester).toMatchObject({
        name: semesterDetails.name,
        starts_at: semesterDetails.starts_at.toString(),
        ends_at: semesterDetails.ends_at.toString(),
      });
    });
  });

  describe('updateSemester', () => {
    it('should throw an error if semester does not exist', async () => {
      const semesterDetails: SemesterDto = {
        name: 'Test Semester',
        starts_at: parseInt(new Date().getTime().toString()),
        ends_at: parseInt(new Date().getTime().toString()),
      };

      await expect(
        semesterService.updateSemester(1, semesterDetails),
      ).rejects.toThrow('Semester not found');
    });

    it('should throw an error if the start date is after the end date', async () => {
      const currentDate: number = parseInt(new Date().getTime().toString());

      await SemesterModel.create({
        name: 'Test Semester',
        created_at: currentDate,
        updated_at: currentDate,
        starts_at: currentDate,
        ends_at: currentDate,
      }).save();

      const semesterDetails: SemesterDto = {
        name: 'Test Semester',
        starts_at: currentDate,
        ends_at: currentDate - 1000,
      };

      await expect(
        semesterService.updateSemester(1, semesterDetails),
      ).rejects.toThrow('Start date must be before end date');
    });

    it('should update a semester', async () => {
      const currentDate: number = parseInt(new Date().getTime().toString());

      await SemesterModel.create({
        name: 'Test Semester',
        created_at: currentDate,
        updated_at: currentDate,
        starts_at: currentDate,
        ends_at: currentDate,
      }).save();

      const semesterDetails: SemesterDto = {
        name: 'Updated Semester',
        starts_at: currentDate,
        ends_at: currentDate + 1000,
      };

      await semesterService.updateSemester(1, semesterDetails);

      const semester = await SemesterModel.findOne({
        where: { id: 1 },
      });

      expect(semester).toBeDefined();

      expect(semester.name).toBe(semesterDetails.name);
    });
  });

  describe('getSemesterById', () => {
    it('should throw an error if semester does not exist', async () => {
      await expect(semesterService.getSemesterById(1)).rejects.toThrow(
        'Semester not found',
      );
    });

    it('should return a semester', async () => {
      await SemesterModel.create({
        name: 'Test Semester',
        created_at: 1_000_000,
        updated_at: 1_000_000,
        starts_at: 1_000_000,
        ends_at: 1_000_000,
      }).save();

      const semester = await semesterService.getSemesterById(1);

      expect(semester).toBeDefined();
      expect(semester).toMatchSnapshot();
    });
  });

  describe('getAllSemesters', () => {
    it('should return all semesters', async () => {
      let semester = await SemesterModel.create({
        name: 'Test Semester',
        created_at: 1_000_000,
        updated_at: 1_000_000,
        starts_at: 1_000_000,
        ends_at: 1_000_000,
      }).save();

      semester = await SemesterModel.findOne({
        where: { id: semester.id },
        relations: ['courses'],
      });

      for (let i = 0; i < 10; i++) {
        const course = await CourseModel.create({
          course_code: 'COSC 499',
          course_name: 'Capstone Project',
          created_at: 1_000_000_000,
          updated_at: 1_000_000_000,
          section_name: '001',
          invite_code: '123',
          semester: semester,
        }).save();

        semester.courses.push(course);
      }
      await semester.save();
      const semesters = await semesterService.getAllSemesters();
      expect(semesters).toBeDefined();

      expect(semesters).toMatchSnapshot();
    });

    it('should return an empty array if no semesters are found', async () => {
      const semesters = await semesterService.getAllSemesters();

      expect(semesters).toHaveLength(0);
    });
  });

  describe('deleteSemester', () => {
    it('should throw an error if semester does not exist', async () => {
      await expect(semesterService.deleteSemester(1)).rejects.toThrow(
        'Semester not found',
      );
    });

    it('should delete a semester', async () => {
      let semester = await SemesterModel.create({
        name: 'Test Semester',
        created_at: 1_000_000,
        updated_at: 1_000_000,
        starts_at: 1_000_000,
        ends_at: 1_000_000,
      }).save();

      await semesterService.deleteSemester(semester.id);

      semester = await SemesterModel.findOne({
        where: { id: semester.id },
      });

      expect(semester).toBeNull();
    });

    it('should delete a semester and ensure courses have null value for semester', async () => {
      let semester = await SemesterModel.create({
        name: 'Test Semester',
        created_at: 1_000_000,
        updated_at: 1_000_000,
        starts_at: 1_000_000,
        ends_at: 1_000_000,
      }).save();

      for (let i = 0; i < 10; i++) {
        await CourseModel.create({
          course_code: 'COSC 499',
          course_name: 'Capstone Project',
          created_at: 1_000_000_000,
          updated_at: 1_000_000_000,
          section_name: '001',
          invite_code: '123',
          semester: semester,
        }).save();
      }

      await semesterService.deleteSemester(semester.id);

      semester = await SemesterModel.findOne({
        where: { id: semester.id },
        relations: ['courses'],
      });

      expect(semester).toBeNull();

      const courses = await CourseModel.find({
        relations: ['semester'],
      });

      expect(courses).toHaveLength(10);

      courses.forEach((course) => {
        expect(course.semester).toBeNull();
      });

      expect(courses).toMatchSnapshot();
    });
  });
});
