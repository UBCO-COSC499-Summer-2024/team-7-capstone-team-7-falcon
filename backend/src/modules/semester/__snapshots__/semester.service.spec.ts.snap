// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SemesterService deleteSemester should delete a semester and ensure courses have null value for semester 1`] = `
[
  CourseModel {
    "course_code": "COSC 499",
    "course_name": "Capstone Project",
    "created_at": "1000000000",
    "id": 1,
    "invite_code": "123",
    "is_archived": false,
    "section_name": "001",
    "semester": null,
    "updated_at": "1000000000",
  },
  CourseModel {
    "course_code": "COSC 499",
    "course_name": "Capstone Project",
    "created_at": "1000000000",
    "id": 2,
    "invite_code": "123",
    "is_archived": false,
    "section_name": "001",
    "semester": null,
    "updated_at": "1000000000",
  },
  CourseModel {
    "course_code": "COSC 499",
    "course_name": "Capstone Project",
    "created_at": "1000000000",
    "id": 3,
    "invite_code": "123",
    "is_archived": false,
    "section_name": "001",
    "semester": null,
    "updated_at": "1000000000",
  },
  CourseModel {
    "course_code": "COSC 499",
    "course_name": "Capstone Project",
    "created_at": "1000000000",
    "id": 4,
    "invite_code": "123",
    "is_archived": false,
    "section_name": "001",
    "semester": null,
    "updated_at": "1000000000",
  },
  CourseModel {
    "course_code": "COSC 499",
    "course_name": "Capstone Project",
    "created_at": "1000000000",
    "id": 5,
    "invite_code": "123",
    "is_archived": false,
    "section_name": "001",
    "semester": null,
    "updated_at": "1000000000",
  },
  CourseModel {
    "course_code": "COSC 499",
    "course_name": "Capstone Project",
    "created_at": "1000000000",
    "id": 6,
    "invite_code": "123",
    "is_archived": false,
    "section_name": "001",
    "semester": null,
    "updated_at": "1000000000",
  },
  CourseModel {
    "course_code": "COSC 499",
    "course_name": "Capstone Project",
    "created_at": "1000000000",
    "id": 7,
    "invite_code": "123",
    "is_archived": false,
    "section_name": "001",
    "semester": null,
    "updated_at": "1000000000",
  },
  CourseModel {
    "course_code": "COSC 499",
    "course_name": "Capstone Project",
    "created_at": "1000000000",
    "id": 8,
    "invite_code": "123",
    "is_archived": false,
    "section_name": "001",
    "semester": null,
    "updated_at": "1000000000",
  },
  CourseModel {
    "course_code": "COSC 499",
    "course_name": "Capstone Project",
    "created_at": "1000000000",
    "id": 9,
    "invite_code": "123",
    "is_archived": false,
    "section_name": "001",
    "semester": null,
    "updated_at": "1000000000",
  },
  CourseModel {
    "course_code": "COSC 499",
    "course_name": "Capstone Project",
    "created_at": "1000000000",
    "id": 10,
    "invite_code": "123",
    "is_archived": false,
    "section_name": "001",
    "semester": null,
    "updated_at": "1000000000",
  },
]
`;

exports[`SemesterService getAllSemesters should return all semesters 1`] = `
[
  SemesterModel {
    "course_count": 10,
    "created_at": "1000000",
    "ends_at": "1000000",
    "id": 1,
    "name": "Test Semester",
    "starts_at": "1000000",
    "updated_at": "1000000",
  },
]
`;

exports[`SemesterService getSemesterById should return a semester 1`] = `
SemesterModel {
  "created_at": "1000000",
  "ends_at": "1000000",
  "id": 1,
  "name": "Test Semester",
  "starts_at": "1000000",
  "updated_at": "1000000",
}
`;
